src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -79,16 +219,59 @@ private static string GetBaseSdkTargetFramework()
 #endif
         }
 
+        /// <summary>
+        /// This will return the current running .NET framework version, based on the version of the assembly that owns
+        /// the 'Object' type. The version number returned can be used to infer other things such as .NET Core / Standard.
+        /// </summary>
+        /// <returns>a string representing the version of the current .NET framework</returns>
         private static string GetRuntimeFrameworkVer()
         {
+#if NET45 || NET46
             Assembly assembly = typeof(Object).GetTypeInfo().Assembly;
 
             AssemblyFileVersionAttribute objectAssemblyFileVer =
                         assembly.GetCustomAttributes(typeof(AssemblyFileVersionAttribute))
                                 .Cast<AssemblyFileVersionAttribute>()
                                 .FirstOrDefault();
-
             return objectAssemblyFileVer != null ? objectAssemblyFileVer.Version : "undefined";
+#elif NETSTANDARD1_3
+            return System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription;
+#else
+#error Unrecognized framework
+            return "unknown";
+#endif
+        }
+
+        /// <summary>
+        /// Runtime information for the underlying OS, should include Linux information here as well.
+        /// </summary>
+        /// <returns>String representing the OS, and any version/patch information reported by that OS</returns>
+        private static string GetRuntimeOsType()
+        {
+#if NET45 || NET46
+            return Environment.OSVersion.VersionString;
```

Comment:
From the method name GetRuntimeOsType() - i expect it to return "windows", 'Linux", "osx" only.

But this seem to be getting version.
we need to double check whether these gives right and useful information.
For eg:, Environment.OSVersion.VersionString gives me "Microsoft Windows NT 6.2.9200.0" and System.Runtime.InteropServices.RuntimeInformation.OSDescription gives me "Microsoft Windows 10.0.16299 " on the same app/machine.

Reply:
Yes, I believe it has to do with the deprecated nature of the Environment.OSVersion object [see here](https://docs.microsoft.com/en-us/dotnet/api/system.environment.osversion?view=netframework-4.7#System_Environment_OSVersion), and in [this document it backs it up by saying](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724832(v=vs.85).aspx):

> ...Applications not manifested for Windows 8.1 or Windows 10 will return the Windows 8 OS version value (6.2). To manifest your applications for Windows 8.1 or Windows 10, refer to Targeting your application for Windows.

Therefore, I think the right way forward is to make use of the System.Runtime.InteropServices.RuntimeInforamtion.OSDescription. Unfortunately, that doesn't exist until .NET Framework 4.7.

Reply:
Instead of going to any of this trouble, I've simplified this to simply attempt to return OS type, and completely remove any notion of OS version (this is a complicated beast in .NET/Windows it seems).